pkg/shares/counter_test.go

```diff
@@ -0,0 +1,74 @@
+package shares_test
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	coretypes "github.com/tendermint/tendermint/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/namespace"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+)
+
+func TestCounterMatchesCompactShareSplitter(t *testing.T) {
+	testCases := []struct {
+		txs []coretypes.Tx
+	}{
+		{txs: []coretypes.Tx{newTx(120)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize - 2)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize - 1)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize + 1)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize), newTx(appconsts.ContinuationCompactShareContentSize - 4)}},
+		{txs: newTxs(1000, 100)},
+		{txs: newTxs(100, 1000)},
```

Comment:
[no change needed] the inputs to this test could also be fuzzed given we just need a random `dataLen`

Reply:
Yeah I can explore that in a follow up. I think these cover the edge cases that come to mind